src/main/kotlin/cacophony/utils/Diagnostics.kt

```diff
@@ -1,9 +1,16 @@
 package cacophony.utils
+import cacophony.token.Token
 
 interface Diagnostics {
```

Comment:
We could add `source` to our logs, or something like that, to know who reported which error

it could be generic type `SourceType : Enum<SourceType>` and we could have initially

```kotlin
enum SourceType {
    LEXER,
    PARSER
}
```